homeassistant/components/homekit_controller/button.py

```diff
@@ -46,17 +46,67 @@ class HomeKitButtonEntityDescription(ButtonEntityDescription):
 }
 
 
+class HomeKitEcobeeClearHoldButton(CharacteristicEntity, ButtonEntity):
+    """Representation of a Button control for Ecobee clear hold request."""
+
+    def __init__(
+        self,
+        conn,
+        info,
+        char,
+    ):
+        """Initialise a HomeKit button control."""
+        self.entity_description = ButtonEntityDescription(
+            key=CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD,
+            name="Clear Hold",
+            device_class=ButtonDeviceClass.UPDATE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+        )
+        super().__init__(conn, info, char)
```

Comment:
I _think_ we can hoist this out of __init__ and put make it a class variable and then completely get rid of the `__init__`.

```suggestion
class HomeKitEcobeeClearHoldButton(CharacteristicEntity, ButtonEntity):
    """Representation of a Button control for Ecobee clear hold request."""

    entity_description = ButtonEntityDescription(
        key=CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD,
        name="Clear Hold",
        device_class=ButtonDeviceClass.UPDATE,
        entity_category=ENTITY_CATEGORY_CONFIG,
    )
```

That said, i'm not 100% on the need to even use an entity description here. AIUI EntityDescriptions are when classes are data driven, and this one isn't. In this case the common pattern in other integrations is to do something like:

```suggestion
class HomeKitEcobeeClearHoldButton(CharacteristicEntity, ButtonEntity):
    """Representation of a Button control for Ecobee clear hold request."""
    
    _attr_name = "Clear Hold"
    _attr_device_class = ButtonDeviceClass.UPDATE
    _attr_entity_category = ENTITY_CATEGORY_CONFIG
```

(In this case you could just use `CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD` or `self._char.type` instead of `self.entity_description.key`).

If this works, can you make the same change to `HomeKitEcobeeFanModeNumber`.

Reply:
I made those changes locally and this approach is working fine.